Topics/Development_Process/Prompt_Engineering.md

```diff
@@ -0,0 +1,77 @@
+
+7. Contextual Information: Provide all necessary information within the prompt, especially if the task requires specific knowledge or constraints​​.
+
+8. Prefixes: Use prefixes to indicate different sections or types of content within your prompt, helping the model understand and categorize the information more effectively​​.
+
+9. Adjust Parameters: Experiment with parameters like max output tokens, temperature, top-K, and top-P to fine-tune the creativity and randomness of the responses​​.
+
+10. Iterative Design: Prompt engineering is rarely perfect on the first try. Iterate on your prompt based on the responses you get, refining until you achieve the desired output​​.
+
+## Writing Prompts - A Sample Scenario
+
+Suppose you are an educator and you want the llm to generate 5 MCQ's for you based on certain text that you provide. You want the output in json format
+This is how a good prompt is structured:
+
+"Context: You are a teacher who wants to create multiple-choice questions based on readings given to the students.
+
+ Example Text: The Tinder Fire was a wildfire that burned 16,309 acres (66.00 km2) of the Coconino National Forest in the U.S. state of Arizona during April and May 2018. 
+
+ Example output: "{\"Question\": \"When did the Tinder Fire occur\", \"Answer\": \"A\", \"A\": \"April and May 2018\", \"B\": \"April and May 2019\", \"C\": \"May and June 2018\", \"D\": \"Jan and Feb 2018\"}"
+
+ Using the following information and nothing else, generate 5 Multiple Choice questions based on the information above and return them in JSON format.
+ Exactly follow the JSON format of the example question below. Each question should be separated by "&&" and nothing else.
+ 
+ Actual Text: {text you want to provide}"
+
+
+The provided prompt is well-structured for the following reasons:
+
+1. Clarity of Task: It clearly defines the context and role of the user, in this case, an educator, setting a clear purpose for the language model's task of generating multiple-choice questions.
+
+2. Specific Example: By including an example of the desired output in JSON format, the prompt sets explicit expectations for the structure and content of the language model's responses, reducing ambiguity.
+
+3. Precise Instructions: The prompt specifies the exact number of questions, the separation character, and the information to be used, guiding the model to produce a well-organized and focused output.
+
+For the similar scenario, this is one example of a bad prompt:
+
+"Context: You are a teacher who wants to create multiple-choice questions based on readings given to the students.
+ Using the following information and nothing else, generate 5 Multiple Choice questions based on the information above and return them in JSON format.
+ Exactly follow the JSON format of the example question below. Each question should be separated by "&&" and nothing else.
```

Comment:
This part says to follow the JSON format of the example question below, but it did not give an example. Perhaps remove this sentence?

Reply:
Good catch, I have removed that sentence which still makes that prompt a bad one. Thank you!